documentation/modules/ROOT/pages/connectors/mongodb.adoc

```diff
@@ -7,112 +9,163 @@ include::../_attributes.adoc[]
+MongoDB's replication mechanism provides redundancy and high availability, and is the preferred way to run MongoDB in production.
+MongoDB connector captures the changes in a replica set or sharded cluster.
+
+A MongoDB _replica set_ consists of a set of servers that all have copies of the same data, and replication ensures that all changes made by clients to documents on the replica set's _primary_ are correctly applied to the other replica set's servers, called _secondaries_.
+MongoDB replication works by having the primary record the changes in its _oplog_ (or operation log), and then each of the secondaries reads the primary's oplog and applies in order all of the operations to their own documents.
+When a new server is added to a replica set, that server first performs an https://docs.mongodb.com/manual/core/replica-set-sync/[initial sync] of all of the databases and collections on the primary, and then reads the primary's oplog to apply all changes that might have been made since it began the initial sync.
+This new server becomes a secondary (and able to handle queries) when it catches up to the tail of the primary's oplog.
 
-The Debezium MongoDB connector uses this same replication mechanism, though it does not actually become a member of the replica set. Just like MongoDB secondaries, however, the connector always reads the oplog of the replica set's primary. And, when the connector sees a replica set for the first time, it looks at the oplog to get the last recorded transaction and then performs an _intial sync_ of the primary's databases and collections. When all the data is copied, the connector then starts reading the oplog from the position it read earlier. Operations in the MongoDB oplog are https://docs.mongodb.com/manual/core/replica-set-oplog/[idempotent], so no matter how many times the operations are applied, they result in the same end state.
+The MongoDB connector uses this same replication mechanism, though it does not actually become a member of the replica set.
```

Comment:
It's a bit unfortunate that this commit does a few things at once (which all are good by themselves). Would have been nice if the wrapping of sentences to new lines had been its own commit (which then can be largely ignored while reviewing), and doing other more critical changes in a separate one (which then is easier to review).

Reply:
Ah, yes, this is really very awkward to review. Sorry about that! The initial commit from Paul captures multiple phases of work on the connector content, so there is really no practical way to separate these different phases now. But I have made a recommendation in the contributor notes (work in progress), to recommend that contributor's avoid this in future.

Reply:
Yes, would be really good to stick to this rule, in particular as we have been talking about this before. Thanks!